src/poetry/installation/chef.py

```diff
@@ -2,28 +2,209 @@
+        package.python_versions = ".".join(str(v) for v in self._env.version_info[:3])
+        for requirement in requirements:
+            dependency = Dependency.create_from_pep_508(requirement)
+            package.add_dependency(dependency)
+
+        pool = Factory.create_pool(self._config)
+        installer = Installer(
+            NullIO(),
+            self._env,
+            package,
+            Locker(self._env.path.joinpath("poetry.lock"), {}),
+            pool,
+            Config.create(),
+            InstalledRepository.load(self._env),
+        )
+        installer.update(True)
+        installer.run()
+
+
 class Chef:
     def __init__(self, config: Config, env: Env) -> None:
+        self._config = config
         self._env = env
         self._cache_dir = (
             Path(config.get("cache-dir")).expanduser().joinpath("artifacts")
         )
 
+    def prepare(
+        self, archive: Path, output_dir: Path | None = None, *, editable: bool = False
+    ) -> Path:
+        if not self._should_prepare(archive):
+            return archive
+
+        if archive.is_dir():
+            tmp_dir = tempfile.mkdtemp(prefix="poetry-chef-")
+
+            return self._prepare(archive, Path(tmp_dir), editable=editable)
+
+        return self._prepare_sdist(archive, destination=output_dir)
+
+    def _prepare(
+        self, directory: Path, destination: Path, *, editable: bool = False
+    ) -> Path:
+        from subprocess import CalledProcessError
+
+        with ephemeral_environment(self._env.python) as venv:
+            env = IsolatedEnv(venv, self._config)
+            builder = ProjectBuilder(
+                directory,
+                python_executable=env.executable,
+                scripts_dir=env.scripts_dir,
+                runner=quiet_subprocess_runner,
+            )
+            env.install(builder.build_system_requires)
+
+            stdout = StringIO()
+            error: Exception | None = None
+            try:
+                with redirect_stdout(stdout):
+                    env.install(
+                        builder.build_system_requires
+                        | builder.get_requires_for_build("wheel")
+                    )
+                    path = Path(
+                        builder.build(
+                            "wheel" if not editable else "editable",
+                            destination.as_posix(),
+                        )
+                    )
+            except BuildBackendException as e:
+                disclaimer = (
+                    "Note: This error originates from the build backend,"
+                    " and is likely not a problem with poetry"
+                    " but the concerning package not supporting PEP 517 builds."
+                    " You can verify this by running"
+                    " 'pip wheel --use-pep517 <concerning package>'."
+                )
+                message_parts = [disclaimer, str(e)]
```

Comment:
Suggest that the disclaimer should come last, not first.  The build error message is quite likely to be a wall of text from gcc or whatever, some people will definitely not bother to scroll back, whatever shows up in the last few lines of a message gets more attention.

(pip agrees)
```
blah blah blah
...
  note: This error originates from a subprocess, and is likely not a problem with pip.
```

Also `concerning package` is awkward.  I realise that the package name isn't currently available to this code - but I'd suggest that it's worth rearranging things so that this error can be explicit about which package poetry is failing on (and pointing the finger at!)